sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -92,10 +98,55 @@ private void checkIfSettingIsRuntime(List<Setting> settings, String name) {
         for (Setting<?, ?> setting : settings) {
             if (setting.settingName().equals(name) && !setting.isRuntime()) {
                 throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
-                        "setting '%s' cannot be set/reset in runtime", name));
+                        "setting '%s' cannot be set/reset at runtime", name));
             }
             checkIfSettingIsRuntime(setting.children(), name);
         }
     }
 
+    private static class ExpressionToSettingNameListVisitor extends AstVisitor<Collection<String>, String> {
```

Comment:
I wonder if it would make sense/be possible to change it so that a List<String> is the context and the visit methods add to it in order to avoid creating new lists all the time